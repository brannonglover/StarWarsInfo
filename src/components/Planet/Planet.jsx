import React from 'react';
import { ListBoxStyle } from '../ListBox.styled';
import { PlanetStyleH3, PlanetStyleP } from './Planet.styled';
import strings from '../../strings';

const Planet = ({ planet }) => {
  return (
    <ListBoxStyle>
      <PlanetStyleH3>{planet.name}</PlanetStyleH3>
      <PlanetStyleP>{strings.POPULATION} - {planet.population}</PlanetStyleP>
      <PlanetStyleP>{strings.TERRAIN} - {planet.terrain}</PlanetStyleP>
    </ListBoxStyle>
  )
}

export default Planet;