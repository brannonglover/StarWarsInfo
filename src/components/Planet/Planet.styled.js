import styled from 'styled-components';

export const PlanetStyle = styled.div`
  padding: 8px 16px;
  background: #1b1b1b;
  margin: 16px 0;
  border-radius: 20px;
`;

export const PlanetStyleP = styled.p`
  margin: 6px 0;
  color: #999;
`;

export const PlanetStyleH3 = styled.h3`
  margin: 10px 0;
  color: #ffff57;
`;