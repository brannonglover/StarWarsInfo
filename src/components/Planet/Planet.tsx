import React from "react";
import { ListBoxStyle } from "../ListBox.styled";
import { PlanetStyleH3, PlanetStyleP } from "./Planet.styled";
import strings from "../../common/strings";
import { IPlanet } from "../../common/types";

interface IPlanetProps {
  planet: IPlanet;
}

const Planet: React.FC<IPlanetProps> = (props: IPlanetProps) => {
  const { planet } = props;

  return (
    <ListBoxStyle>
      <PlanetStyleH3>{planet.name}</PlanetStyleH3>
      <PlanetStyleP>
        {strings.POPULATION} - {planet.population}
      </PlanetStyleP>
      <PlanetStyleP>
        {strings.TERRAIN} - {planet.terrain}
      </PlanetStyleP>
    </ListBoxStyle>
  );
};

export default Planet;
