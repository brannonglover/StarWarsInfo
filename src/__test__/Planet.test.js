import React from 'react';
import { cleanup, render } from '@testing-library/react';
import Planet from '../components/Planet/Planet';

afterEach(cleanup);

test('Display Planet component', () => {
  const planet = {
    name: "Some Planet",
    population: "1000",
    terrain: "rough"
  }
  const { asFragment } = render(<Planet planet={planet} />);

  expect(asFragment()).toMatchSnapshot();
});