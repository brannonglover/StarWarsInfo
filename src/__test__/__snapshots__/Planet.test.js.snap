// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Planet component 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ hkBXND"
  >
    <h3
      class="sc-dlfnbm cwBEuk"
    >
      Some Planet
    </h3>
    <p
      class="sc-gsTCUz ggAsXq"
    >
      Population - 1000
    </p>
    <p
      class="sc-gsTCUz ggAsXq"
    >
      Terrain - rough
    </p>
  </div>
</DocumentFragment>
`;
